File: backend/app/schemas/document.py
from pydantic import BaseModel, EmailStr, field_validator, model_validator
from typing import Optional, List, Dict, Any
from datetime import datetime
from uuid import UUID


class DocumentAuthor(BaseModel):
    user_id: Optional[int] = None
    username: Optional[str] = None
    email: Optional[EmailStr] = None


class DocumentRelation(BaseModel):
    type: str  # references, related, parent, child
    target_slug: str
    description: Optional[str] = None


class DocumentBase(BaseModel):
    slug: str
    title: str
    category: Optional[str] = None
    content_summary: Optional[str] = None
    published: bool = True
    tags: List[str] = []
    relations: List[DocumentRelation] = []
    extra_metadata: Dict[str, Any] = {}


class DocumentCreate(DocumentBase):
    file_path: Optional[str] = None
    author_id: Optional[int] = None


class DocumentUpdate(BaseModel):
    slug: Optional[str] = None
    title: Optional[str] = None
    category: Optional[str] = None
    content_summary: Optional[str] = None
    published: Optional[bool] = None
    tags: Optional[List[str]] = None
    relations: Optional[List[DocumentRelation]] = None
    extra_metadata: Optional[Dict[str, Any]] = None


class DocumentResponse(DocumentBase):
    id: UUID
    file_path: str
    author_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/document.ts
export interface DocumentAuthor {
  user_id?: number
  username?: string
  email?: string
}

export interface DocumentRelation {
  type: 'references' | 'related' | 'parent' | 'child'
  target_slug: string
  description?: string
}

export interface Document {
  id?: string  // 可选，文件系统文档可能没有 id
  slug: string
  title: string
  file_path: string
  category?: string | null
  description?: string | null
  content_summary?: string | null  // 兼容旧字段
  content_preview?: string | null  // 文件系统文档的预览
  author_id?: number | null
  published: boolean
  tags?: string[]  // 可选，文件系统文档可能为空数组
  relations?: DocumentRelation[]  // 可选
  extra_metadata?: Record<string, any>  // 可选
  frontmatter?: Record<string, any>  // 文件系统文档的前置元数据
  content?: string  // 完整内容（仅在详情页时存在）
  created_at: string
  updated_at: string
}

export interface DocumentContent {
  content: string
}


